pkg/composableschemadsl/compiler/translator.go

```diff
@@ -22,6 +23,8 @@ type translationContext struct {
 	mapper           input.PositionMapper
 	schemaString     string
 	skipValidate     bool
+	existingNames    []string
```

Comment:
Should this be a `*mapz.Set` up and down the stack?

Reply:
Likely